### ISSUE TITLE
example basic-compute-shader  fails on Apple M1 Silicon

### ISSUE THREAD

**Issue by phelluy** [vulkano-rs/vulkano#1730]
# System Config

*   Version of vulkano: 0.26
*   OS: macOS 11.6
*   GPU (the selected PhysicalDevice): Apple M1 (type: IntegratedGpu)
*   GPU Driver: Vulkan version  1.2.189 (based on MoltenVK)
*   Minimal complete `main.rs` file that demonstrates the issue: the example is taken from the example ```basic-compute-shader``` taken unchanged from the github repository

# Issue

I am able to build everything, but the test fails on a MacBook Pro (13-inch, M1, 2020) with the following messages:

```
./target/debug/basic-compute-shader

Using device: Apple M1 (type: IntegratedGpu)
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ExtensionRestrictionNotMet(ExtensionRestrictionError { extension: "khr_portability_subset", restriction: RequiresInstanceExtension("khr_get_physical_device_properties2") })', examples/src/bin/basic-compute-shader.rs:70:6
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by Eliah-Lakhin** [comment]
@phelluy I merged a Pull Request from @Rua with the fix, but I wan't able to test it, I don't have Mac. Can you test it please and let us know if it work? And please reopen the Issue if the bug persisted on the latest `master`.

**Comment by Rua** [comment]
Actually, I think this issue may have been closed in error (my mistake). It is true that the PR should fix the problem that `khr_get_physical_device_properties2` is required even on Vulkan 1.1 and above. However, if you are using a Vulkan 1.0 instance, then this extension will be required and the examples will still fail. So the examples need to be adjusted so that they enable this extension on Vulkan 1.0 if it's available.

**Comment by phelluy** [comment]
Now the basic-compute-shader compiles and run in debug. But in release I cannot compile:

cargo build --release              
   Compiling examples v0.1.0 (/Users/helluy/vulkano/examples)
error[E0428]: the name `Data` is defined multiple times
  --> examples/src/bin/shader-types-sharing.rs:83:9
   |
83 |         vulkano_shaders::shader! {
   |         ^^^^^^^^^^^^^^^^^^^^^^^^
   |         |
   |         `Data` redefined here
   |         previous definition of the type `Data` here

**Comment by Rua** [comment]
I can reproduce that issue for the `shader-types-sharing` example on Linux, so it's not related to Mac or your original issue. I'll open a new one.

**Comment by Eliah-Lakhin** [comment]
@phelluy, @Rua I believe this issue is now fixed(in #1765). Can you confirm, please?

**Comment by phelluy** [comment]
I confirm that now everything compile also in release mode. And the above tests and examples pass.
thanks !
